warehouse/warehouse_test.go

```diff
@@ -0,0 +1,400 @@
+
+				t.Run(tc.name+" with "+subTC.name, func(t *testing.T) {
+					pgResource, err := resource.SetupPostgres(pool, t)
+					require.NoError(t, err)
+
+					webPort, err := kithelper.GetFreePort()
+					require.NoError(t, err)
+
+					ctx, stopServer := context.WithCancel(context.Background())
+
+					port, err := strconv.Atoi(pgResource.Port)
+					require.NoError(t, err)
+
+					c := config.New()
+					c.Set("WAREHOUSE_JOBS_DB_HOST", pgResource.Host)
+					c.Set("WAREHOUSE_JOBS_DB_PORT", port)
+					c.Set("WAREHOUSE_JOBS_DB_USER", pgResource.User)
+					c.Set("WAREHOUSE_JOBS_DB_PASSWORD", pgResource.Password)
+					c.Set("WAREHOUSE_JOBS_DB_DB_NAME", pgResource.Database)
+					c.Set("Warehouse.mode", tc.warehouseMode)
+					c.Set("Warehouse.runningMode", subTC.runningMode)
+					c.Set("Warehouse.webPort", webPort)
+
+					a := NewApp(mockApp, c, logger.NOP, stats.Default, &bcConfig.NOOP{}, filemanager.New)
+					err = a.Setup(ctx)
+					require.NoError(t, err)
+
+					g, gCtx := errgroup.WithContext(ctx)
+					g.Go(func() error {
+						return a.Start(gCtx)
+					})
+					g.Go(func() error {
+						defer stopServer()
+
+						serverURL := fmt.Sprintf("http://localhost:%d", webPort)
+
+						require.Eventually(t, func() bool {
+							resp, err := http.Get(fmt.Sprintf("%s/health", serverURL))
+							if err != nil {
+								return false
+							}
+							defer func() {
+								httputil.CloseResponse(resp)
+							}()
+
+							return resp.StatusCode == http.StatusOK
+						},
+							time.Second*10,
+							time.Second,
+						)
+						return nil
+					})
+					require.NoError(t, g.Wait())
+				})
+			}
+		}
+	})
+	t.Run("incompatible postgres", func(t *testing.T) {
+		pgResource, err := resource.SetupPostgres(pool, t, postgres.WithTag("9-alpine"))
+		require.NoError(t, err)
+
+		port, err := strconv.Atoi(pgResource.Port)
+		require.NoError(t, err)
+
+		c := config.New()
+		c.Set("WAREHOUSE_JOBS_DB_HOST", pgResource.Host)
+		c.Set("WAREHOUSE_JOBS_DB_PORT", port)
+		c.Set("WAREHOUSE_JOBS_DB_USER", pgResource.User)
+		c.Set("WAREHOUSE_JOBS_DB_PASSWORD", pgResource.Password)
+		c.Set("WAREHOUSE_JOBS_DB_DB_NAME", pgResource.Database)
+
+		a := NewApp(mockApp, c, logger.NOP, stats.Default, &bcConfig.NOOP{}, filemanager.New)
+		err = a.Setup(context.Background())
+		require.EqualError(t, err, "setting up database: warehouse Service needs postgres version >= 10. Exiting")
+	})
+	t.Run("postgres down", func(t *testing.T) {
+		c := config.New()
+		c.Set("WAREHOUSE_JOBS_DB_HOST", "localhost")
+		c.Set("WAREHOUSE_JOBS_DB_PORT", "5432")
+		c.Set("WAREHOUSE_JOBS_DB_USER", "ubuntu")
+		c.Set("WAREHOUSE_JOBS_DB_PASSWORD", "ubuntu")
+		c.Set("WAREHOUSE_JOBS_DB_DB_NAME", "ubuntu")
+
+		a := NewApp(mockApp, c, logger.NOP, stats.Default, &bcConfig.NOOP{}, filemanager.New)
+		err = a.Setup(context.Background())
+		require.ErrorContains(t, err, "setting up database: could not check compatibility:")
+	})
+	t.Run("without env vars", func(t *testing.T) {
```

Comment:
Why without env vars? It looks like we're passing them :thinking:

Reply:
I will update the test context. It was for `without warehouse env vars`. In that case, we are using `misc.GetConnectionString`